swagger/src/test/scala/org/http4s/rho/swagger/SwaggerSupportSuite.scala

```diff
@@ -177,30 +198,37 @@ class SwaggerSupportSuite extends FunSuite {
 
     val r = Request[IO](GET, Uri(path = Uri.Path.unsafeFromString("/swagger.json")))
 
-    val json = parse(RRunner(service).checkOk(r)).getOrElse(???)
+    val jsonF = RRunner(service)
+      .checkOk(r)
+      .map(parse)
+      .flatMap(_.map(_.pure[IO]).getOrElse(IO.raiseError(new Throwable("Expected Right value"))))
 
-    val security: List[Json] = (json \\ "security").flatMap(_.asArray).flatten
+    val securityF = jsonF.map(json => (json \\ "security").flatMap(_.asArray).flatten)
 
-    assert(
-      security.contains(
-        Json.obj(
-          "bye" := List("hello")
+    assertIOBoolean(
+      securityF.map(
+        _.contains(
+          Json.obj(
+            "bye" := List("hello")
+          )
         )
       )
     )
 
-    assert(
-      security.contains(
-        Json.obj(
-          "hello" := List("bye")
+    assertIOBoolean(
+      securityF.map(
+        _.contains(
+          Json.obj(
+            "hello" := List("bye")
+          )
         )
       )
     )
 
-    val summary = (json \\ "summary").flatMap(_.asString)
+    val summaryF = jsonF.map(json => (json \\ "summary").flatMap(_.asString))
 
-    assert(summary.contains("Bye"))
-    assert(summary.contains("Hello"))
+    assertIOBoolean(summaryF.map(_.contains("Bye")))
+    assertIOBoolean(summaryF.map(_.contains("Hello")))
```

Comment:
I don't think these are quite right.  I think the first 3 `assertIOBoolean` are completely ignored since their `IO`'s aren't going to be evaluated.

Reply:
You are 100% right. I've fixed that.